feature_engine/transformation/arcsin.py

```diff
@@ -0,0 +1,178 @@
+    feature_names_in_=_feature_names_in_docstring,
+    n_features_in_=_n_features_in_docstring,
+    fit=_fit_not_learn_docstring,
+    fit_transform=_fit_transform_docstring,
+    inverse_transform=_inverse_transform_docstring,
+)
+class ArcsinTransformer(BaseNumericalTransformer):
+    """
+    The ArcsinTransformer() applies the arcsin transformation to numerical variables.
+
+    The ArcsinTransformer() only works with numerical variables between 0 and +1.
+    If a variable contains a value outside of this range, the transformer will
+    raise an error.
+
+    A list of variables can be passed as an argument. Alternatively, the transformer
+    will automatically select and transform all numerical variables.
+
+    More details in the :ref:`User Guide <arcsin>`.
+
+    Parameters
+    ----------
+    {variables}
+
+    Attributes
+    ----------
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {inverse_transform}
+
+    transform:
+        Apply the arcsin transformation.
+
+    """
+
+    def __init__(
+        self, variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn parameters.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features].
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+        """
+
+        # check input dataframe
+        X = super()._fit_from_varlist(X)
+
+        # check if the variables is in the correct range
+        if ((X[self.variables_] < 0) | (X[self.variables_] > 1)).any().any():
+            raise ValueError(
+                "Some variables contain values outside the possible range"
+                "can't apply arcsin transformation"
+            )
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Apply the arcsin transformation.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame of shape = [n_samples, n_features]
+            The data to be transformed.
+
+        Returns
+        -------
+        X_new: pandas dataframe
+            The dataframe with the transformed variables.
+        """
+
+        # check input dataframe and if class was fitted
+        X = super().transform(X)
+
+        # check if the variables is in the correct range
+        if ((X[self.variables_] < 0) | (X[self.variables_] > 1)).any().any():
+            raise ValueError(
+                "Some variables contain values outside the possible"
+                "range for arcsin, can't apply arcsin transformation"
```

Comment:
Can we modify the error message like the one in fit please?

Reply:
done